File: Go/len_of_last_word/58_test.go
package lenoflastword

import (
	"fmt"
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestLenOfLastWord(t *testing.T) {
	tests := [...]struct {
		s    string
		want int
	}{
		{s: "Hello World", want: 5},
		{s: "   fly me   to   the moon  ", want: 4},
		{s: "luffy is still joyboy", want: 6},
	}

	for i, tt := range tests {
		t.Run(fmt.Sprintf("case %d", i+1), func(t *testing.T) {
			got := lengthOfLastWord(tt.s)
			assert.Equal(t, tt.want, got)
		})
	}
}

File: TypeScript/len_of_last_word/58.test.ts
import lengthOfLastWord from "./58";

interface tt {
  s: string;
  want: number;
}

const tests: tt[] = [
  { s: "Hello World", want: 5 },
  { s: "   fly me   to   the moon  ", want: 4 },
  { s: "luffy is still joyboy", want: 6 },
];

describe("length of last word", () => {
  tests.forEach((t, i) => {
    test(`case ${i + 1}`, () => {
      const result = lengthOfLastWord(t.s);
      expect(result).toBe(t.want);
    });
  });
});
